src/traversal.js

```diff
@@ -36,18 +39,21 @@ var enterNode = function(node) {
 var exitNode = function(node) {
   var data = getData(node);
   var lastVisitedChild = data.lastVisitedChild;
+  var lastChild = node.lastChild;
 
-  if (node.lastChild === lastVisitedChild) {
+  if (lastChild === lastVisitedChild) {
     return;
   }
 
-  while (node.lastChild !== lastVisitedChild) {
-    node.removeChild(node.lastChild);
-  }
+  while (lastChild !== lastVisitedChild) {
+    var key = getKey(lastChild);
+    if (key) {
+      registerChild(node, key, undefined);
```

Comment:
We would want to delete the key off the mapping so that its size doesn't keep growing. I think I'd like to re-evaluate this change later though, I'm not certain how big of a problem it really is.